### PR TITLE
ArgoCD CRD

### PR DESCRIPTION
**Pull request by disposab1e**
Fetched all the information form:
https://github.com/argoproj/argo-helm/blob/master/charts/argo-cd/values.yaml
https://github.com/argoproj/argo-helm/tree/master/charts/argo-cd

operator-sdk generate openapi 👍 
operator-sdk generate k8s 👍 

Not sure this is all fine, first lines of go and first go operator work, so please forgive me when it's not perfekt :-)

### CODE REVIEW

File: pkg/apis/argoproj/v1alpha1/argocd_types.go [hunk 1]
```diff
@@ -9,9 +10,207 @@ import (
...
+	// Describe a container image
+	Image ImageStruct `json:"image"`
+}
+
+type IngressStruct struct {
+	// Enable ingress
+	Enabled bool `json:"enabled"`
+	// Annotations for ingress object
+	// +optional
+	Annotations map[string]string `json:"anntotations,omitempty"`
```
Comment (disposab1e) [inline pkg/apis/argoproj/v1alpha1/argocd_types.go]: not sure with this datatype

File: pkg/apis/argoproj/v1alpha1/argocd_types.go [hunk 2]
```diff
@@ -9,9 +10,207 @@ import (
...
+	ServicePortHttp int `json:"servicePortHttp"`
+	// Service port for Dex Server GRPC
+	ServicePortGrpc int `json:"servicePortGrpc"`
+	// Describe a container image
+	Image ImageStruct `json:"image"`
+	// Describe a container image
+	InitImage ImageStruct `json:"initImage"`
+	// Additional volume mounts
+	// +optional
+	VolumeMounts []corev1.VolumeMount `json:"volumeMounts,omitempty"`
```
Comment (disposab1e) [inline pkg/apis/argoproj/v1alpha1/argocd_types.go]: not sure with this datatype

File: pkg/apis/argoproj/v1alpha1/argocd_types.go [hunk 3]
```diff
@@ -9,9 +10,207 @@ import (
...
+	// Additional volumes
+	// +optional
+	Volumes []corev1.VolumeSource `json:"volumes,omitempty"`
```
Comment (disposab1e) [inline pkg/apis/argoproj/v1alpha1/argocd_types.go]: not sure with this datatype

File: pkg/apis/argoproj/v1alpha1/argocd_types.go [hunk 4]
```diff
@@ -9,9 +10,207 @@ import (
...
+type RepoServerStruct struct {
+	// Container port for repo server
+	ContainerPort int `json:"containerPort"`
+	// Service port for repo server
+	ServicePort int `json:"servicePort"`
+	// Describe a container image
+	Image ImageStruct `json:"image"`
+	// Additional volume mounts
+	// +optional
+	VolumeMounts []corev1.VolumeMount `json:"volumeMounts,omitempty"`
```
Comment (disposab1e) [inline pkg/apis/argoproj/v1alpha1/argocd_types.go]: not sure with this datatype

File: pkg/apis/argoproj/v1alpha1/argocd_types.go [hunk 5]
```diff
@@ -9,9 +10,207 @@ import (
...
+	// Additional volumes
+	// +optional
+	Volumes []corev1.VolumeSource `json:"volumes,omitempty"`
```
Comment (disposab1e) [inline pkg/apis/argoproj/v1alpha1/argocd_types.go]: not sure with this datatype

File: pkg/apis/argoproj/v1alpha1/argocd_types.go [hunk 6]
```diff
@@ -9,9 +10,207 @@ import (
...
+	ExtraArgs []string `json:"extraArgs,omitempty"`
+	// Additional volume mounts
+	// +optional
+	VolumeMounts []corev1.VolumeMount `json:"volumeMounts,omitempty"`
+	// Additional volumes
+	// +optional
+	Volumes []corev1.VolumeSource `json:"volumes,omitempty"`
+	// Annotations for the server deployment
+	// +optional
+	Annotations map[string]string `json:"anntotations,omitempty"`
```
Comment (disposab1e) [inline pkg/apis/argoproj/v1alpha1/argocd_types.go]: not sure with this datatype

File: pkg/apis/argoproj/v1alpha1/argocd_types.go [hunk 7]
```diff
@@ -9,9 +10,207 @@ import (
...
+	ServiceType corev1.ServiceType `json:"serviceType"`
+	// Add additional arguments
+	// +optional
+	ExtraArgs []string `json:"extraArgs,omitempty"`
+	// Additional volume mounts
+	// +optional
+	VolumeMounts []corev1.VolumeMount `json:"volumeMounts,omitempty"`
+	// Additional volumes
+	// +optional
+	Volumes []corev1.VolumeSource `json:"volumes,omitempty"`
```
Comment (disposab1e) [inline pkg/apis/argoproj/v1alpha1/argocd_types.go]: not sure with this datatype

File: pkg/apis/argoproj/v1alpha1/argocd_types.go [hunk 8]
```diff
@@ -9,9 +10,207 @@ import (
...
+	// Describe a container image
+	Image ImageStruct `json:"image"`
+	// Service Type string describes ingress methods for a service
+	ServiceType corev1.ServiceType `json:"serviceType"`
+	// Add additional arguments
+	// +optional
+	ExtraArgs []string `json:"extraArgs,omitempty"`
+	// Additional volume mounts
+	// +optional
+	VolumeMounts []corev1.VolumeMount `json:"volumeMounts,omitempty"`
```
Comment (disposab1e) [inline pkg/apis/argoproj/v1alpha1/argocd_types.go]: not sure with this datatype

**Comment by jmckind** [comment]
Thanks @disposab1e, I am reviewing this.

**Comment by disposab1e** [comment]
cool! Do you have a source of information for me regarding "validating" and "defaulting"?

**Comment by jmckind** [comment]
Thank you for this PR. However it adds many properties and while they may be used in the future, I would prefer a more phased approach to adding additional properties to the CRD. In addition, we need to get some tests written for the logic we have now.

**Pull request closed without merging**